staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -66,6 +66,14 @@ type Allocator struct {
 	// access to this map must be synchronized.
 	availableCounters map[string]counterSets
 	mutex             sync.RWMutex
+	// allocationAttemptsByClaim collects the number of different permutations
+	// of devices that was attempted before succeeding or failing to allocate
+	// devices for a claim.
+	// The key in the map is the index of the claim in claimsToAllocate and
+	// the value is the number of permutations.
+	// Access to the map must be syncronized.
+	allocationAttemptsByClaim map[int]int64
```

Comment:
I wonder whether we really need to track this by claim. I expect the question of when to call `incAllocationAttempt` (see below) to become easier to answer when we count leaf nodes in the overall search tree, and the semantic also becomes simpler.